plugins/modules/cloud/xenserver/xenserver_guest.py

```diff
@@ -104,56 +103,137 @@
     - All parameters are case sensitive.
     - Removing or detaching existing disks of VM is not supported.
-    - 'Required parameters per entry:'
-    - ' - C(size_[tb,gb,mb,kb,b]) (integer): Disk storage size in specified unit. VM needs to be shut down to reconfigure this parameter.'
-    - 'Optional parameters per entry:'
-    - ' - C(name) (string): Disk name. You can also use C(name_label) as an alias.'
-    - ' - C(name_desc) (string): Disk description.'
-    - ' - C(sr) (string): Storage Repository to create disk on. If not specified, will use default SR. Cannot be used for moving disk to other SR.'
-    - ' - C(sr_uuid) (string): UUID of a SR to create disk on. Use if SR name is not unique.'
+    - Entries are required to have either a I(size) or one of I(size_[tb,gb,mb,kb,b]) parameters.
+    - VM needs to be shut down to reconfigure disk size.
     type: list
     elements: dict
     aliases: [ disk ]
+    suboptions:
+      size:
+        description:
+        - 'Disk size with unit. Unit must be: C(b), C(kb), C(mb), C(gb), C(tb). VM needs to be shut down to reconfigure this parameter.'
+        type: str
+      size_b:
+        description:
+        - Disk size in bytes.
+        type: str
+      size_kb:
+        description:
+        - Disk size in kilobytes.
+        type: str
+      size_mb:
+        description:
+        - Disk size in megabytes.
+        type: str
+      size_gb:
+        description:
+        - Disk size in gigabytes.
+        type: str
+      size_tb:
+        description:
+        - Disk size in terabytes.
+        type: str
+      name:
+        description:
+        - Disk name.
+        type: str
+        aliases: [ name_label ]
+      name_desc:
+        description:
+        - Disk description.
+        type: str
+      sr:
+        description:
+        - Storage Repository to create disk on. If not specified, will use default SR. Cannot be used for moving disk to other SR.
+        type: str
+      sr_uuid:
+        description:
+        - UUID of a SR to create disk on. Use if SR name is not unique.
+        type: str
   cdrom:
     description:
     - A CD-ROM configuration for the VM.
     - All parameters are case sensitive.
-    - 'Valid parameters are:'
-    - ' - C(type) (string): The type of CD-ROM, valid options are C(none) or C(iso). With C(none) the CD-ROM device will be present but empty.'
-    - ' - C(iso_name) (string): The file name of an ISO image from one of the XenServer ISO Libraries (implies C(type: iso)).
-          Required if C(type) is set to C(iso).'
     type: dict
+    suboptions:
+      type:
+        description:
+        - The type of CD-ROM. With C(none) the CD-ROM device will be present but empty.
+        type: str
+        choices: [ none, iso ]
+      iso_name:
+        description:
+          - 'The file name of an ISO image from one of the XenServer ISO Libraries (implies I(type): C(iso)).'
+          - Required if I(type) is set to C(iso).
```

Comment:
Fix indentation. There are two extra spaces compared to the rest of the documentation.